sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala

```diff
@@ -69,8 +66,7 @@ class DefaultSource extends HadoopFsRelationProvider with DataSourceRegister {
 
 private[sql] class JSONRelation(
     val inputRDD: Option[RDD[String]],
-    val samplingRatio: Double,
-    val primitivesAsString: Boolean,
+    val configOptions: JSONOptions,
```

Comment:
`JSONRelation` has a constructor argument `parameters` that contains all data source settings. We can create this `JSONOptions` inside `JSONRelation`.